lib/query/query_payments.rb

```diff
@@ -144,8 +144,10 @@ def self.select_to_filter_search(npo_id, query)
       campaign_search = campaign_and_child_query_as_raw_string
       inner_donation_search = inner_donation_search.where("donations.campaign_id IN (#{campaign_search})", id: query[:campaign_id])
     end
+
+    inner_supporters_search = Qexpr.new.select('supporters.*').from('supporters').where('NOT supporters.deleted')
     expr = Qexpr.new.select('payments.id').from('payments')
-          .inner_join('supporters', "supporters.id=payments.supporter_id")
+          .inner_join(inner_supporters_search.as('supporters'), "supporters.id=payments.supporter_id")
```

Comment:
Does it make sense with what you said about not including deleted supporters on the result?

Reply:
Was it previously including deleted supporters in the result?

Reply:
Oops, I remember this now: you need to have payments from deleted supporters as well so this change should not be made.

After all, consider a charge that was made and then supporter was deleted. The nonprofit would want that record to show as a payment (yes, we should have a better way to handling "deleted" supporters).

Please put a comment there explaining why we don't want only deleted supporters there, otherwise we may make this mistake again :smile: